src/ode_solver/ode_solver_factory.cpp

```diff
@@ -125,6 +98,132 @@ void ODESolverFactory<dim,real,MeshType>::display_error_ode_solver_factory(Param
     std::abort();
 }
 
+template <int dim, typename real, typename MeshType>
+std::shared_ptr<ODESolverBase<dim,real,MeshType>> ODESolverFactory<dim,real,MeshType>::create_RungeKuttaODESolver(std::shared_ptr< DGBase<dim,real,MeshType> > dg_input)
+{
+    dealii::ConditionalOStream pcout(std::cout, dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD)==0);
+
+    std::shared_ptr<RKTableauBase<dim,real,MeshType>> rk_tableau = create_RKTableau(dg_input);
+   
+
+    const int n_rk_stages = dg_input->all_parameters->ode_solver_param.n_rk_stages;
+    using ODEEnum = Parameters::ODESolverParam::ODESolverEnum;
+    ODEEnum ode_solver_type = dg_input->all_parameters->ode_solver_param.ode_solver_type;
+    if (ode_solver_type == ODEEnum::runge_kutta_solver) {
+
+        pcout << "Creating Runge Kutta ODE Solver";
+        if (n_rk_stages == 1){
+            pcout << " with 1 stage..." << std::endl;
+            return std::make_shared<RungeKuttaODESolver<dim,real,1,MeshType>>(dg_input,rk_tableau);
+        }
+        if (n_rk_stages == 2){
+            pcout << " with 2 stages..." << std::endl;
+            return std::make_shared<RungeKuttaODESolver<dim,real,2,MeshType>>(dg_input,rk_tableau);
+        }
+        if (n_rk_stages == 3){
+            pcout << " with 3 stages..." << std::endl;
+            return std::make_shared<RungeKuttaODESolver<dim,real,3,MeshType>>(dg_input,rk_tableau);
+        }
+        if (n_rk_stages == 4){
+            pcout << " with 4 stages..." << std::endl;
+            return std::make_shared<RungeKuttaODESolver<dim,real,4,MeshType>>(dg_input,rk_tableau);
+        }
+        else{
+            pcout << "but number of stages was unclear. Aborting..." << std::endl;
+            std::abort();
+            return nullptr;
+        }
+    }
+    if (ode_solver_type == ODEEnum::rrk_explicit_solver){
+        if constexpr(dim==1){
+            const bool use_collocated_nodes = dg_input->all_parameters->use_collocated_nodes;
+            using PDEEnum = Parameters::AllParameters::PartialDifferentialEquation;
+            const PDEEnum pde_type = dg_input->all_parameters->pde_type;
+            const bool use_inviscid_burgers = (pde_type == PDEEnum::burgers_inviscid);
+            if (use_collocated_nodes && use_inviscid_burgers){
+                pcout << "Creating Relaxation Runge Kutta ODE Solver";
+                if (n_rk_stages == 1){
+                    pcout << " with 1 stage..." << std::endl;
+                    return std::make_shared<RRKExplicitODESolver<dim,real,1,MeshType>>(dg_input,rk_tableau);
+                }
+                if (n_rk_stages == 2){
+                    pcout << " with 2 stages..." << std::endl;
+                    return std::make_shared<RRKExplicitODESolver<dim,real,2,MeshType>>(dg_input,rk_tableau);
+                }
```

Comment:
same comment as before (combine all if statements)

Reply:
I've implemented the same `pcout` restructuring.